engine/consensus/compliance/engine.go

```diff
@@ -168,7 +171,13 @@ func (e *Engine) Ready() <-chan struct{} {
 	}
 	e.lm.OnStart(func() {
 		e.unit.Launch(e.loop)
```

Comment:
refactoring the code to comply with the new convention for `ReadyDoneAware` would be great; probably as a separate PR; using `ComponentManager` would be great